include/mxnet/ndarray.h

```diff
@@ -770,6 +770,12 @@ class NDArray {
    */
   NDArray Reorder2Default() const;
 
+    /*
+   * This creates a new NDArray using f32 with the reordered data.
+   * It doesn't affect the data of the original NDArray.
+   */
+  NDArray Reorder2DefaultFp32() const;
```

Comment:
Adding dtype specific interface looks very ad-hoc

Reply:
OK, will change it to Reorder2DefaultFloatFormat()